contracts/protocol/pool/PoolApeStaking.sol

```diff
@@ -0,0 +1,392 @@
+
+            // 1, send borrow part to xTokenAddress
+            if (stakingInfo.borrowAmount > 0) {
+                ValidationLogic.validateFlashloanSimple(apeReserve);
+                IPToken(apeReserveCache.xTokenAddress).transferUnderlyingTo(
+                    nTokenAddress,
+                    stakingInfo.borrowAmount
+                );
+            }
+
+            // 2, send cash part to xTokenAddress
+            if (stakingInfo.cashAmount > 0) {
+                apeCoin.safeTransferFrom(
+                    msg.sender,
+                    nTokenAddress,
+                    stakingInfo.cashAmount
+                );
+            }
+
+            // 3, deposit bayc or mayc pool
+            INToken nToken = INToken(nTokenAddress);
+            for (uint256 index = 0; index < _nfts.length; index++) {
+                require(
+                    nToken.ownerOf(_nfts[index].tokenId) == msg.sender,
+                    Errors.NOT_THE_OWNER
+                );
+            }
+            nTokenApeStaking.depositApeCoin(_nfts);
+
+            // 4, deposit bakc pool
+            IERC721 bakcContract = nTokenApeStaking.getBAKC();
+            for (uint256 index = 0; index < _nftPairs.length; index++) {
+                require(
+                    nToken.ownerOf(_nftPairs[index].mainTokenId) == msg.sender,
+                    Errors.NOT_THE_OWNER
+                );
+
+                bakcContract.safeTransferFrom(
+                    msg.sender,
+                    nTokenAddress,
+                    _nftPairs[index].bakcTokenId
+                );
+            }
+            nTokenApeStaking.depositBAKC(_nftPairs);
+            //transfer BAKC back for user
+            for (uint256 index = 0; index < _nftPairs.length; index++) {
+                bakcContract.safeTransferFrom(
+                    nTokenAddress,
+                    msg.sender,
+                    _nftPairs[index].bakcTokenId
+                );
+            }
+        }
+
+        // 5 set sape as collateral
+        setSApeUseAsCollateral(msg.sender, true);
+
+        // 6 mint debt token
+        if (stakingInfo.borrowAmount > 0) {
+            BorrowLogic.executeBorrow(
+                ps._reserves,
+                ps._reservesList,
+                ps._usersConfig[msg.sender],
+                DataTypes.ExecuteBorrowParams({
+                    asset: address(apeCoin),
+                    user: msg.sender,
+                    onBehalfOf: msg.sender,
+                    amount: stakingInfo.borrowAmount,
+                    referralCode: 0,
+                    releaseUnderlying: false,
+                    reservesCount: ps._reservesCount,
+                    oracle: ADDRESSES_PROVIDER.getPriceOracle(),
+                    priceOracleSentinel: ADDRESSES_PROVIDER
+                        .getPriceOracleSentinel()
+                })
+            );
+        }
+
+        // 7 check user hf
+        require(
```

Comment:
is there a chance that HF could drop below one here?

Reply:
Normally no. but yubo said it might be if we set a very high LTV for sApe.